utils.py

```diff
@@ -662,41 +663,39 @@ def __missing__(self, key):
 
 
 class hashabledict(dict):
```

Comment:
I understand the motivation for a hashable dict. It is too bad Python doesn't have `frozendict` like `frozenset`.

But the implementation here is O(n) to compare or hash. Given that, a simpler representation is:

    class hdict(dict):
        def __hash__(self): return 1